osfs/os_posix.go

```diff
@@ -1,5 +1,5 @@
-//go:build !plan9 && !windows && !js
-// +build !plan9,!windows,!js
+//go:build !plan9 && !windows && !wasm
+// +build !plan9,!windows,!wasm
```

Comment:
We may need to revisit some of the build conditions across the files within `osfs`. I suspect here we would also want `!js`.

In other places where we have `js`, we may also want to add `wasm` - for example `os_js.go`.